build_projects/dotnet-host-build/TestTargets.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.Cli.Build.Framework;
+
+using static Microsoft.DotNet.Cli.Build.FS;
+using static Microsoft.DotNet.Cli.Build.Framework.BuildHelpers;
+using static Microsoft.DotNet.Cli.Build.Utils;
+using Microsoft.DotNet.Cli.Build;
+
+namespace Microsoft.DotNet.Host.Build
+{
+    public class TestTargets
+    {
+        private static string s_testPackageBuildVersionSuffix = "<buildversion>";
+
+        public static readonly string[] TestProjects = new[]
+        {
+            "HostActivationTests"
+        };
+
+        [Target(
+            nameof(PrepareTargets.Init),
+            nameof(RestoreTestAssets),
+            nameof(RestoreTests),
+            nameof(BuildTests),
+            nameof(RunTests))]
+        public static BuildTargetResult Test(BuildTargetContext c) => c.Success();
+
+        [Target]
+        public static BuildTargetResult RestoreTestAssets(BuildTargetContext c)
+        {
+            CleanBinObj(c, Path.Combine(Dirs.RepoRoot, "TestAssets"));
+
+            DotNetCli.Stage0.Restore(
+                    "--verbosity", "verbose",
+                    "--fallbacksource", Dirs.CorehostLocalPackages,
+                    "--fallbacksource", Dirs.CorehostDummyPackages,
+                    "--disable-parallel")
+                .WorkingDirectory(Path.Combine(Dirs.RepoRoot, "TestAssets"))
+                .Execute()
+                .EnsureSuccessful();
+
+            return c.Success();
+        }
+
+        [Target]
+        public static BuildTargetResult RestoreTests(BuildTargetContext c)
+        {
+            CleanBinObj(c, Path.Combine(Dirs.RepoRoot, "test"));
+
+            DotNetCli.Stage0.Restore("--verbosity", "verbose")
+                .WorkingDirectory(Path.Combine(Dirs.RepoRoot, "test"))
+                .Execute()
+                .EnsureSuccessful();
+            return c.Success();
+        }
+
+        [Target]
+        public static BuildTargetResult BuildTests(BuildTargetContext c)
+        {
+            var dotnet = DotNetCli.Stage0;
+
+            var configuration = c.BuildContext.Get<string>("Configuration");
+
+            foreach (var testProject in TestProjects)
+            {
+                c.Info($"Building tests: {testProject}");
+                dotnet.Build("--configuration", configuration)
+                    .WorkingDirectory(Path.Combine(Dirs.RepoRoot, "test", testProject))
+                    .Execute()
+                    .EnsureSuccessful();
+            }
+            return c.Success();
+        }
+
+        [Target]
+        public static BuildTargetResult RunTests(BuildTargetContext c)
+        {
+            var dotnet = DotNetCli.Stage0;
```

Comment:
Same here. And the funny thing is that you actually use DotNETCLI.Stage0 in the error message formatting.